enhancements/sig-architecture/34-work-executor-group/README.md

```diff
@@ -0,0 +1,147 @@
+# Work Executor Group
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management-io/website/)
+
+## Summary
+
+The proposal adds a new field `executorGroupName` to the spec of `ManifestWork` 
+api, to separate the execution of a set of `ManifestWork` resources to the 
+dedicated work agents. The default/legacy work agent will be the default
+executor of the `ManifestWork` resource if its `executorGroupName` is not
+specified.
+
+## Motivation
+
+### Controlling availability (blast-radius) of the work agents
+
+Currently, for each of the managed clusters we will deploy the work agents
+in a single deployment resource which is sharing the availability via "leader-
+election", so there's merely one instance of work agent taking effect at a time.
+Under this pattern, we can't break down the availability of the work agent into
+finer units so that:
+
+- During the upgrade, we can progressively reduce the impact from the agent's 
```

Comment:
During an upgrade, wouldn't we roll over a single deployment pod at a time, so that the outage would be negligible?  If not, is something wrong with the deployment structure or perhaps the lease configuration (perhaps graceful release is disabled?)

Reply:
i removed the agent instance splitting in the update. by leveraging the impersonation, we can switch the identities from one instance on the fly.